docs: add key encryption docs (#28369)
supabase/supabase: apps/docs/docs/ref/self-hosting-analytics/introduction.mdx

**We recommend using the BigQuery backend for production environments as it offers better scaling and querying/debugging experiences.**

### Rotate Encryption Keys Regularly

The Logflare server uses the a Base64 encryption key set on the `LOGFLARE_DB_ENCRYPTION_KEY` environment variable to perform encryption at rest for sensitive database columns.

To perform encryption key rotation, move the retired key to the `LOGFLARE_DB_ENCRYPTION_KEY_RETIRED` environment variable, and replace the `LOGFLARE_DB_ENCRYPTION_KEY` environement variable with the new key. Perform a server restart and check `info` logs for the migration to be detected and performed.

> Once migration is complete, the retired key can be safely removed.

Reviewer: ```suggestion
Once migration is complete, you safely remove the retired key.
```
